doc/guides/dts/intro.rst

```diff
@@ -339,6 +339,107 @@ interrupts
 .. _Devicetree Specification release v0.3:
    https://www.devicetree.org/specifications/
 
+.. highlight:: none
+
+.. _dt-writing-property-values:
+
+Writing property values
+***********************
+
+This section describes how to write property values in DTS format. The property
+types in the table below are described in detail in :ref:`dt-bindings`.
+
+Some specifics are skipped in the interest of keeping things simple; if you're
+curious about details, see the devicetree specification.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 4 4
+
+   * - Property type
+     - How to write
+     - Example
+
+   * - string
+     - Double quoted
+     - ``a-string = "hello, world!";``
+
+   * - int
+     - between angle brackets (``<`` and ``>``)
+     - ``an-int = "<1>";``
+
+   * - boolean
+     - for ``true``, with no value (for ``false``, use ``/delete-property/``)
+     - ``my-true-boolean;``
+
+   * - array
+     - between angle brackets (``<`` and ``>``), separated by spaces
+     - ``foo = <0xdeadbeef 1234 0>;``
+
+   * - uint8-array
+     - in hexadecimal *without* leading ``0x``, between square brackets (``[`` and ``]``).
+     - ``a-byte-array = [00 01 ab];``
+
+   * - string-array
+     - separated by commas
+     - ``a-string-array = "string one", "string two", "string three";``
+
+   * - phandle
+     - between angle brackets (``<`` and ``>``)
+     - ``a-phandle = <&mynode>;``
+
+   * - phandles
+     - between angle brackets (``<`` and ``>``), separated by spaces
+     - ``some-phandles = <&mynode0 &mynode1 &mynode2>;``
+
+   * - phandle-array
+     - between angle brackets (``<`` and ``>``), separated by spaces
+     - ``a-phandle-array = <&mynode0 1 2 &mynode1 3 4>;``
+
+Additional notes on the above:
+
+- Boolean properties are true if present. They should not have a value.
+  A boolean property is only false if it is completely missing in the DTS.
+
+- The ``foo`` property value above has three *cells* with values 0xdeadbeef, 1234,
+  and 0, in that order. Note that hexadecimal and decimal numbers are allowed and
+  can be intermixed. Since Zephyr transforms DTS to C sources, it is not
+  necessary to specify the endianness of an individual cell here.
+
+- 64-bit integers are written as two 32-bit cells in big-endian order. The value
+  0xaaaa0000bbbb1111 would be written ``<0xaaaa0000 0xbbbb1111>``.
+
+- The ``a-byte-array`` property value is the three bytes 0x00, 0x01, and 0xab, in
+  that order.
+
+- Parentheses, arithmetic operators, and bitwise operators are allowed. The
+  ``bar`` property contains a single cell with value 64::
+
+  	bar = <2 * (1 << 5)>;
```

Comment:
If I'm not wrong, the whole expression needs to be enclosed in parenthesis, i.e. `bar = <(2 * (1 << 5))>`